src/data/polynomial.lean

```diff
@@ -2354,12 +2354,26 @@ derivative_C
 by refine finsupp.sum_add_index _ _; intros;
 simp only [add_mul, zero_mul, C_0, C_add, C_mul]
 
+/-- The formal derivative of polynomials, as additive homomorphism. -/
+noncomputable def derivative_hom (R : Type*) [comm_semiring R] : polynomial R →+ polynomial R :=
```

Comment:
Why this has to be `noncomputable`? In case of a `comm_ring R`, could you please add a `linear_map` version?

Reply:
Oops, I don't need that in this file (-;

Reply:
I added the the linear map version, under the assumption that `R` is a commutative *ring*.